### DISCUSSION TITLE
HowTo: Use input values as useEffect dependencies

### DISCUSSION THREAD

**Discussion by andrewgadziksonos** [react-hook-form/react-hook-form#2713] [Q&A]
Hello friends 👋 

I'm trying to use `watch` or `getValues` to get the current form values and use those as `useEffect` dependencies, but I cant seem to trigger my `useEffect` callback.  Here's a codesandbox example,

https://codesandbox.io/s/react-hook-form-useform-template-forked-jm57c

Any help is much appreciated!

**Comment by bluebill1049** [comment]
why not just remove that `useEffect`?

**Reply by andrewgadziksonos** [reply]
I was trying to simulate an API call inside that `useEffect` with `setTimeout` - The use case is that I have a form, and I need to make a call to an API when input values change and are valid.  Maybe I'm going about this all wrong 🤷‍♂️ .  I appreciate the help so far!

**Comment by andrewgadziksonos** [comment]
I was able to get this working!  Turns out I fat fingered some variable names.  Opps!

Checkout the codesandbox for a complete example.

**Reply by bluebill1049** [reply]
which part? maybe we can put some dev warning in the next build.

**Reply by andrewgadziksonos** [reply]
i had `const { FirstName, LastName } = watch()` when I needed `const { firstName, lastName } = watch()`.

**Reply by bluebill1049** [reply]
hmmm, that would be tricky...
